extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/ImperativeTypeBundle.java

```diff
@@ -0,0 +1,61 @@
+package io.quarkus.mongodb.panache.deployment;
+
+import io.quarkus.mongodb.panache.PanacheMongoEntity;
+import io.quarkus.mongodb.panache.PanacheMongoEntityBase;
+import io.quarkus.mongodb.panache.PanacheMongoRepository;
+import io.quarkus.mongodb.panache.PanacheMongoRepositoryBase;
+import io.quarkus.mongodb.panache.PanacheQuery;
+import io.quarkus.mongodb.panache.PanacheUpdate;
+import io.quarkus.mongodb.panache.runtime.MongoOperations;
+
+public class ImperativeTypeBundle implements TypeBundle {
+    @Override
+    public ByteCodeType entity() {
+        return new ByteCodeType(PanacheMongoEntity.class);
+    }
+
+    @Override
+    public ByteCodeType entityBase() {
+        return new ByteCodeType(PanacheMongoEntityBase.class);
+    }
+
+    @Override
+    public ByteCodeType entityBaseCompanion() {
+        throw new UnsupportedOperationException("Companions are not supported in Java.");
```

Comment:
Feels weird to have this method defined in the Java base interface if it's only meant for Kotlin implementations. Is it called by the abstract processor? If not, it should probably belong in a Kotlin subinterface.

Reply:
Well, the idea was to have a single interface.  Generally the java side would never call that because it doesn't make sense but it means we can simply pass around `TypeBundle` references and not need to worry about subtypes.  I had actually thought about bumping the definition of those methods to default methods and then the java implementations can simply ignore them.  But I did have a case once early on when I was experimenting with this approach where I had passed in the wrong type and got the java type defs instead and this exception revealed that error quite quickly.